src/reducers/starWarsReducer.js

```diff
@@ -1,10 +1,38 @@
-import /* we need our action types here*/ "../actions";
+import {
+  FETCH_API_START,
+  FETCH_API_SUCCESS,
+  FETCH_API_ERROR
+} from "../actions";
+
 const initialState = {
-  characters: []
+  characters: [],
+  error: " ",
+  isLoading: false
+
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
+    case FETCH_API_START:
+      return {
+        ...state,
+        error: " ",
+        isLoading: true
+      };
+    case FETCH_API_SUCCESS:
+      return {
+        ...state,
+        error: " ",
+        isLoading: false,
+        characters: action.payload.results
+      };
+    case FETCH_API_ERROR:
+      return {
+        ...state,
+        error: "",
```

Comment:
```suggestion
        error: action.payload.message,
```
Ok, so when the api call fails, you want to populate this error state right? Otherwise why have it? I drilled it down for you, every error you get back from the server is going to be different, you need to look at it and see if it has any sort of message for you, and if not I would populate your error state with a custom error message